app/components/session-offerings.js

```diff
@@ -3,28 +3,153 @@ import Ember from 'ember';
 import layout from '../templates/components/session-offerings';
 import { translationMacro as t } from "ember-i18n";
 
-export default Ember.Component.extend({
-  store: Ember.inject.service(),
-  i18n: Ember.inject.service(),
-  layout: layout,
+const { Component, computed, inject, isPresent, RSVP } = Ember;
+const { service } = inject;
+const { alias, oneWay } = computed;
+const { all } = RSVP;
+
+export default Component.extend({
+  saving: false,
+
+  store: service(),
+
+  i18n: service(),
+
+  layout,
+
   classNames: ['session-offerings'],
+
   session: null,
+
   placeholderValue: t('sessions.titleFilterPlaceholder'),
-  offerings: Ember.computed.oneWay('session.offerings'),
+
+  offerings: oneWay('session.offerings'),
+
   newButtonTitle: t('general.add'),
-  newOfferings: [],
+
+  offeringEditorOn: false,
+
+  cohorts: alias('session.course.cohorts'),
+
   actions: {
-    add: function(){
-      var self = this;
-      var offering = this.get('store').createRecord('offering', {
-        session: self.get('session'),
-        startDate: moment().hour(8).minute(0).second(0).format(),
-        endDate: moment().hour(9).minute(0).second(0).format(),
+    addSingleOffering({ startDate, endDate, room, learnerGroups, instructors, instructorGroups }) {
+      const store = this.get('store');
+      const session = this.get('session');
+      const offering = store.createRecord('offering', { session, startDate, endDate, room });
+
+      offering.save().then((offering) => {
+        offering.get('learnerGroups').then((offeringlearnerGroups) => {
+          learnerGroups.forEach((learnerGroup) => {
+            offeringlearnerGroups.pushObject(learnerGroup);
+            learnerGroup.save();
```

Comment:
Just a reminder to do all of these changes and then save the offering at the end instead of saving each learner group / instructor / instructor group.  That should be a a single put request which should be much faster.